Compiled from "ListenableFutureAdapter.java"
class org.springframework.util.concurrent.ListenableFutureAdapter$1 implements org.springframework.util.concurrent.ListenableFutureCallback<S> {
  final org.springframework.util.concurrent.SuccessCallback val$successCallback;

  final org.springframework.util.concurrent.FailureCallback val$failureCallback;

  final org.springframework.util.concurrent.ListenableFutureAdapter this$0;

  org.springframework.util.concurrent.ListenableFutureAdapter$1(org.springframework.util.concurrent.ListenableFutureAdapter, org.springframework.util.concurrent.SuccessCallback, org.springframework.util.concurrent.FailureCallback);
    Code:
       0: aload_0
       1: aload_1
       2: putfield      #1                  // Field this$0:Lorg/springframework/util/concurrent/ListenableFutureAdapter;
       5: aload_0
       6: aload_2
       7: putfield      #2                  // Field val$successCallback:Lorg/springframework/util/concurrent/SuccessCallback;
      10: aload_0
      11: aload_3
      12: putfield      #3                  // Field val$failureCallback:Lorg/springframework/util/concurrent/FailureCallback;
      15: aload_0
      16: invokespecial #4                  // Method java/lang/Object."<init>":()V
      19: return

  public void onSuccess(S);
    Code:
       0: aconst_null
       1: astore_2
       2: aload_1
       3: ifnull        48
       6: aload_0
       7: getfield      #1                  // Field this$0:Lorg/springframework/util/concurrent/ListenableFutureAdapter;
      10: aload_1
      11: invokevirtual #5                  // Method org/springframework/util/concurrent/ListenableFutureAdapter.adaptInternal:(Ljava/lang/Object;)Ljava/lang/Object;
      14: astore_2
      15: goto          48
      18: astore_3
      19: aload_3
      20: invokevirtual #7                  // Method java/util/concurrent/ExecutionException.getCause:()Ljava/lang/Throwable;
      23: astore        4
      25: aload_0
      26: aload         4
      28: ifnull        36
      31: aload         4
      33: goto          37
      36: aload_3
      37: invokevirtual #8                  // Method onFailure:(Ljava/lang/Throwable;)V
      40: return
      41: astore_3
      42: aload_0
      43: aload_3
      44: invokevirtual #8                  // Method onFailure:(Ljava/lang/Throwable;)V
      47: return
      48: aload_0
      49: getfield      #2                  // Field val$successCallback:Lorg/springframework/util/concurrent/SuccessCallback;
      52: aload_2
      53: invokeinterface #10,  2           // InterfaceMethod org/springframework/util/concurrent/SuccessCallback.onSuccess:(Ljava/lang/Object;)V
      58: return
    Exception table:
       from    to  target type
           6    15    18   Class java/util/concurrent/ExecutionException
           6    15    41   Class java/lang/Throwable

  public void onFailure(java.lang.Throwable);
    Code:
       0: aload_0
       1: getfield      #3                  // Field val$failureCallback:Lorg/springframework/util/concurrent/FailureCallback;
       4: aload_1
       5: invokeinterface #11,  2           // InterfaceMethod org/springframework/util/concurrent/FailureCallback.onFailure:(Ljava/lang/Throwable;)V
      10: return
}
